packages/calypso-e2e/src/lib/components/editor-nav-sidebar-component.ts

```diff
@@ -0,0 +1,83 @@
+import { Page, FrameLocator } from 'playwright';
+import envVariables from '../../env-variables';
+import { NavbarComponent } from '.';
+
+const panel = 'div.wpcom-block-editor-nav-sidebar-nav-sidebar__container';
+const selectors = {
+	sidebarButton: `button.wpcom-block-editor-nav-sidebar-toggle-sidebar-button__button`,
+	exitLink: `${ panel } a.wpcom-block-editor-nav-sidebar-nav-sidebar__home-button`,
+};
+
+/**
+ * Represents an instance of the WordPress.com Editor's navigation sidebar.
+ * The component is available only in the Desktop viewport.
+ */
+export class EditorNavSidebarComponent {
+	private page: Page;
+	private frameLocator: FrameLocator;
+
+	/**
+	 * Constructs an instance of the component.
+	 *
+	 * @param {Page} page The underlying page.
+	 * @param {FrameLocator} frameLocator Locator of the editor iframe.
+	 */
+	constructor( page: Page, frameLocator: FrameLocator ) {
+		this.page = page;
+		this.frameLocator = frameLocator;
+	}
+
+	/**
+	 * Opens the sidebar if not already open.
+	 */
+	async openSidebar(): Promise< void > {
+		if ( envVariables.VIEWPORT_NAME === 'mobile' ) {
+			return;
+		}
+
+		if ( ! ( await this.sidebarIsOpen() ) ) {
+			const locator = this.frameLocator.locator( selectors.sidebarButton );
+			await locator.click();
+		}
+	}
+
+	/**
+	 * Exits the editor.
+	 *
+	 * For Desktop viewports, the sidebar is used to exit.
+	 * For Mobile viewports, My Sites button is used to exit.
+	 */
+	async exitEditor(): Promise< void > {
+		// There are three different places to return to,
+		// depending on how the editor was entered.
+		const navigationPromise = Promise.race( [
+			this.page.waitForNavigation( { url: '**/home/**' } ),
+			this.page.waitForNavigation( { url: '**/posts/**' } ),
+			this.page.waitForNavigation( { url: '**/pages/**' } ),
+		] );
+		const actions: Promise< unknown >[] = [ navigationPromise ];
+
+		if ( envVariables.VIEWPORT_NAME === 'mobile' ) {
+			// Mobile viewports do not use a sidebar.
+			const navbarComponent = new NavbarComponent( this.page );
+			actions.push( navbarComponent.clickMySites() );
```

Comment:
I feel bad saying this because I know we're trying to take things out of the EditorPage, but this handling feels a little bit strange to me here. Per the single responsibility principle, I think we should try to make each component only responsible for itself, or components it "contains". So, the editor nav sidebar should only contain actions that are in that sidebar.

That practically in this case would mean pulling this logic into the EditorPage, which is a bummer, but I think that makes the most sense in terms of code hygiene!

What do you think?

Reply:
I've moved the logic up to the parent/calling method, `GutenbergEditorPage.exitEditor` in 4122ed003d9348d80efb3ad926838b891533454b.